core/field_number.js

```diff
@@ -119,9 +119,8 @@ Blockly.FieldNumber.prototype.doClassValidation_ = function(opt_newValue) {
   newValue = newValue.replace(/O/ig, '0');
   // Strip out thousands separators.
   newValue = newValue.replace(/,/g, '');
-
-  // Clean up number.
-  var n = Number(newValue || 0);
+  var hex = newValue.match(/^0xX+$/g);
```

Comment:
This doesn't do what you think it does. This match is looking for start of the line followed by '0x' followed by 1 or more 'X' characters followed by the end of the line. So it would match '0xXXXXXXXXXXXX' but not '0xabc'.